tests/test_run.py

```diff
@@ -0,0 +1,455 @@
+CLI_RUN_NAME = "my_cli_run"
+
+PATH_TO_CTX_MGR_RUN = f"{rh.Run.LOCAL_RUN_PATH}/{CTX_MGR_RUN}"
+
+RUN_FILES = (
+    rh.Run.INPUTS_FILE,
+    rh.Run.RESULT_FILE,
+    rh.Run.RUN_CONFIG_FILE,
+    ".out",
+    ".err",
+)
+
+
+# ------------------------- FUNCTION RUN ----------------------------------
+
+
+@pytest.mark.clustertest
+@pytest.mark.runstest
+def test_read_fn_stdout(cpu_cluster, submitted_run):
+    """Reads the stdout for the Run."""
+    fn_run = cpu_cluster.get_run(submitted_run)
+    stdout = fn_run.stdout()
+    pprint(stdout)
+    assert stdout
+
+
+@pytest.mark.clustertest
+@pytest.mark.runstest
+def test_load_run_result(cpu_cluster, submitted_run):
+    """Load the Run created above directly from the cluster."""
+    # Note: Run only exists on the cluster (hasn't yet been saved to RNS).
+    func_run = cpu_cluster.get_run(submitted_run)
+    assert func_run.result() == 3
+
+
+@pytest.mark.clustertest
+@pytest.mark.runstest
+def test_get_or_call_from_cache(summer_func, submitted_run):
+    """Cached version of synchronous run - if already completed return the result, otherwise run and wait for
+    completion before returning the result."""
+    # Note: In this test since we already ran the function, it should return the result without re-running
+    run_output = summer_func.get_or_call(name_run=submitted_run)
+    assert run_output == 3
+
+
+@pytest.mark.clustertest
+@pytest.mark.runstest
+def test_get_or_call_no_cache(summer_func):
+    """Cached version of synchronous run - if already completed return the result, otherwise run and wait for
+    completion before returning the result."""
+    # Note: In this test since we do not have a run with this name, it should first execute the function
+    # before returning its result
+    run_output = summer_func.get_or_call(name_run="another_sync_run", a=1, b=2)
+    assert run_output == 3
+
+
+@pytest.mark.clustertest
+@pytest.mark.runstest
+def test_invalid_fn_sync_run(summer_func, cpu_cluster):
+    """Test error handling for invalid function Run. The function expects to receive integers but
+    does not receive any. No serialized result should be saved down on the cluster for this Run, and an error
+    should be thrown via Ray."""
+    import ray
+
+    try:
+        summer_func.get_or_call(name_run="invalid_run")
+    except ray.exceptions.RayTaskError as e:
+        assert (
+            str(e.args[0])
+            == "summer() missing 2 required positional arguments: 'a' and 'b'"
+        )
+
+
+@pytest.mark.clustertest
+@pytest.mark.runstest
+def test_invalid_fn_async_run(summer_func):
+    """Test error handling for invalid function Run. The function expects to receive integers but
+    does not receive any. No result should be saved down on the cluster for this Run, and the Run object returned
+    should have a status of `ERROR`."""
+    run_obj = summer_func.get_or_run(name_run="invalid_async_run")
+
+    assert run_obj.refresh().status == rh.RunStatus.ERROR
```

Comment:
Would be good to grab the exception itself here and check it, like the test above.